src/app.js

```diff
@@ -455,10 +455,13 @@ var run = function () {
 
   function editorOnChange () {
     var input = editor.getValue()
+    // save contents
+    editor.setCacheFileContent(input)
```

Comment:
The content will be saved each time the content is modified ?
I would perhaps just add a save button next to the compile button, and save the content automatically in a *.save.sol file every X minutes. So users could retrieve the content if something goes wrong.
Cause i guess saving a 500 lines contract every time will slowdown everything.